### PR TITLE
iOS READMEs - Cloud and Portal

### PR DESCRIPTION
**Pull request by vquach2404**
Updated READMEs, image names, and sample descriptions.

### CODE REVIEW

File: arcgis-ios-sdk-samples/Cloud & Portal/Authenticate with OAuth/README.md [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
+Authenticate with ArcGIS Online (or your own portal) using OAuth2 to access secured resources (such as private web maps or layers).
 
-Your app may need to access items that are only shared with authorized users. For example, your organization may host private data layers or feature services that are only accessible by verified users. You may also need to take advantage of premium ArcGIS Online services, such as geocoding or routing, that require a named user login.
+![Login screen](authenticate-with-oauth1.png) ![Map view after authentication](authenticate-with-oauth2.png)
 
-![Screenshot 1](image1.png) ![Screenshot 2](image2.png)
+## Use case
+Your app may need to access items that are only shared with authorized users. For example, your organization may host private data layers or feature services that are only accessible by verified users. You may also need to take advantage of premium ArcGIS Online services, such as geocoding or routing, that require a named user login.
```
Comment (philium) [inline arcgis-ios-sdk-samples/Cloud & Portal/Authenticate with OAuth/README.md]:
```suggestion
## Use case

Your app may need to access items that are only shared with authorized users. For example, your organization may host private data layers or feature services that are only accessible by verified users. You may also need to take advantage of premium ArcGIS Online services, such as geocoding or routing, that require a named user login.
```

File: arcgis-ios-sdk-samples/Cloud & Portal/Authenticate with OAuth/README.md [hunk 2]
```diff
@@ -1,31 +1,34 @@
...
-- `AGSAuthenticationManager`
-- `AGSPortal`
- 
+ * AGSAuthenticationManager
+ * AGSOAuthConfiguration
+ * AGSPortalItem
+
+## Additional information
+
+The workflow presented in this sample works for all SAML based enterprise (IWA, PKI, Okta, etc.) & social (facebook, google, etc.) identity providers for ArcGIS Online or Portal. For more information, see the topic [Set up enterprise logins].(https://doc.arcgis.com/en/arcgis-online/administer/enterprise-logins.htm).
```
Comment (philium) [inline arcgis-ios-sdk-samples/Cloud & Portal/Authenticate with OAuth/README.md]:
```suggestion
The workflow presented in this sample works for all SAML based enterprise (IWA, PKI, Okta, etc.) & social (Facebook, Google, etc.) identity providers for ArcGIS Online or Portal. For more information, see the topic [Set up enterprise logins].(https://doc.arcgis.com/en/arcgis-online/administer/enterprise-logins.htm).
```

File: arcgis-ios-sdk-samples/Cloud & Portal/Integrated Windows Authentication/README.md [hunk 3]
```diff
@@ -1,33 +1,45 @@
...
 
 ## Use case
 
-IWA, which is built into Microsoft Internet Information Server (IIS), works well for intranet applications, but isn't always practical for internet apps.
+Your organization might use Integrated Windows Authentication (IWA) to secure ArcGIS Enterprise. This can be useful because the same credentials used to log into your work computer and network can be used to authenticate with ArcGIS. IWA is built into Microsoft Internet Information Server (IIS) and works well for intranet applications but isn't always practical for internet apps.
 
 ## How to use the sample
 
-1. Enter the URL to your IWA-secured portal in the text field above the Search Private button.
+1. Enter the URL to your IWA-secured portal in the text field above the Search Secure button.
```
Comment (philium) [inline arcgis-ios-sdk-samples/Cloud & Portal/Integrated Windows Authentication/README.md]: 👍🏻 That was a last minute change and it looks like I forgot to update the screenshot as well. 😞

File: arcgis-ios-sdk-samples/Cloud & Portal/Integrated Windows Authentication/README.md [hunk 4]
```diff
@@ -1,33 +1,45 @@
...
 2. Tap either the Search Public button (which will search for portals on www.arcgis.com) or Search Secure button (which will search your IWA-secured portal), to search for web maps stored on the portal.
 3. If you tap Search Secure, you will be prompted for a username (including domain, such as username@DOMAIN or domain\username) and password.
 4. If you authenticate successfully, portal item results will display in the table view.
 5. Tap a web map item to display it in a map view.
 
+## Relevant API
+* AGSPortal
```
Comment (philium) [inline arcgis-ios-sdk-samples/Cloud & Portal/Integrated Windows Authentication/README.md]:
```suggestion
## Relevant API

* AGSPortal
```

File: arcgis-ios-sdk-samples/Cloud & Portal/List portal group users/README.md [hunk 5]
```diff
@@ -1,13 +1,29 @@
...
 
-This sample demonstrates how to show the users in a portal group.
+Shows a list of users in a portal group.
 
-![](image1.png)
+![List Portal Group Users](list-portal-group-users.png)
 
-## How it works
+## Use case
+This portal information can be used to provide a customized UI experience for the user. For example, you can show a thumbnail next to their username in the header of an application to indicate that they are currently logged in. Additionally, apps such as Collector and Explorer use this functionality to integrate with Portal.
```
Comment (philium) [inline arcgis-ios-sdk-samples/Cloud & Portal/List portal group users/README.md]:
```suggestion
## Use case

This portal information can be used to provide a customized UI experience for the user. For example, you can show a thumbnail next to their username in the header of an application to indicate that they are currently logged in. Additionally, apps such as Collector and Explorer use this functionality to integrate with Portal.
```

File: arcgis-ios-sdk-samples/Cloud & Portal/List portal group users/README.md [hunk 6]
```diff
@@ -1,13 +1,29 @@
...
+
+## How to use the sample
+When launched, this sample displays a list of users in the Runtime Group portal. Notice that along with displaying each user's name and picture, their description is displayed as well.
```
Comment (philium) [inline arcgis-ios-sdk-samples/Cloud & Portal/List portal group users/README.md]:
```suggestion
## How to use the sample

When launched, this sample displays a list of users in the Runtime Group portal. Notice that along with displaying each user's name and picture, their description is displayed as well. 
```

File: arcgis-ios-sdk-samples/Cloud & Portal/List portal group users/README.md [hunk 7]
```diff
@@ -1,13 +1,29 @@
...
+This portal information can be used to provide a customized UI experience for the user. For example, you can show a thumbnail next to their username in the header of an application to indicate that they are currently logged in. Additionally, apps such as Collector and Explorer use this functionality to integrate with Portal.
+
+## How to use the sample
+When launched, this sample displays a list of users in the Runtime Group portal. Notice that along with displaying each user's name and picture, their description is displayed as well. 
 
-The sample initializes an `AGSPortal` object. A group on this portal is retrieved using the `findGroups(with:completion:)` method. From this group, all users are fetched using the `fetchUsers(completion:)` method. The method returns an array of user names. An `AGSPortalUser` object is created for each user name using the `init(portal:username:)` initializer. Each of these portalUser object is loaded and the user information is shown in a table view.
+## How it works
 
+1. Initialize an `AGSPortal` object.
+2. Retrieve a group from this portal using `findGroups(with:completion:)` which  returns an array of user names.
```
Comment (philium) [inline arcgis-ios-sdk-samples/Cloud & Portal/List portal group users/README.md]:
```suggestion
2. Retrieve a group from this portal using `findGroups(with:completion:)`, which returns an array of user names.
```

File: arcgis-ios-sdk-samples/Cloud & Portal/List portal group users/README.md [hunk 8]
```diff
@@ -1,13 +1,29 @@
...
+3. Obtain all the users in this group by using `fetchUsers(completion:)`.
+4. Create an `AGSPortalUser` object for each user name using `init(portal:username:)`.
 
+## Relevant API
+* AGSPortal
```
Comment (philium) [inline arcgis-ios-sdk-samples/Cloud & Portal/List portal group users/README.md]:
```suggestion
## Relevant API

* AGSPortal
```

File: arcgis-ios-sdk-samples/Cloud & Portal/List portal group users/README.md [hunk 9]
```diff
@@ -1,13 +1,29 @@
...
+* AGSPortalGroup
+* AGSPortalUser
 
+## About the data
+This sample displays users who are a part of the Runtime Group portal.
```
Comment (philium) [inline arcgis-ios-sdk-samples/Cloud & Portal/List portal group users/README.md]:
```suggestion
## About the data

This sample displays users who are a part of the Runtime Group portal.
```

File: arcgis-ios-sdk-samples/Cloud & Portal/List portal group users/README.md [hunk 10]
```diff
@@ -1,13 +1,29 @@
...
 
+## Tags
+account, cloud and portal, description, profile, user, username
```
Comment (philium) [inline arcgis-ios-sdk-samples/Cloud & Portal/List portal group users/README.md]:
```suggestion
## Tags

account, cloud and portal, description, profile, user, username
```

File: arcgis-ios-sdk-samples/Cloud & Portal/Search for webmap by keyword/README.md [hunk 11]
```diff
@@ -1,17 +1,27 @@
 # Search for webmap by keyword
 
-Shows how to search for webmaps within a portal using a keyword.
+Find webmap portal items by using a search term.
 
-![](image1.png)
+![Search for Webmap by Keyword](search-for-webmap-by-keyword.png)
+
+## Use case
+Portals can contain many portal items and at times, you may wish to query the portal to find what you're looking for. In this example, we search for webmap portal items using a text search.
```
Comment (philium) [inline arcgis-ios-sdk-samples/Cloud & Portal/Search for webmap by keyword/README.md]:
```suggestion
## Use case

Portals can contain many portal items and, at times, you may wish to query the portal to find what you're looking for. In this example, we search for webmap portal items using a text search.
```

File: arcgis-ios-sdk-samples/Cloud & Portal/Search for webmap by keyword/README.md [hunk 12]
```diff
@@ -1,17 +1,27 @@
...
 
 ## How to use the sample
 
-Input a keyword into the search field. Results load automatically as you type. Tap a result to show the webmap in a map view. 
+Enter search terms into the search bar. Once the search is complete, a list is populated with the resultant webmaps. Tap on a webmap to set it to the map view. Scrolling to the bottom of the webmap recycler view will get more results.
```
Comment (philium) [inline arcgis-ios-sdk-samples/Cloud & Portal/Search for webmap by keyword/README.md]: Recycler view is a platform-specific thing (Android, I think).

```suggestion
Enter search terms into the search bar. Once the search is complete, a list is populated with the resultant webmaps. Tap on a webmap to set it to the map view..
```

File: arcgis-ios-sdk-samples/Cloud & Portal/Search for webmap by keyword/README.md [hunk 13]
```diff
@@ -1,17 +1,27 @@
...
-To search for webmaps in a `AGSPortal` matching a keyword:
-
-1. Create a `AGSPortal` object and load it.
-2. Create a `AGSPortalQueryParameters` object using `init(forItemsOf:withSearch:)`. Pass `.webMap` as the type and use the search field text to build a search query. Note that webmaps authored prior to July 2nd, 2014, are not supported, so you can also limit the query to only return maps published after that date.
+1. Create an `AGSPortal` and load it.
+2. Create an `AGSPortalQueryParameters` object using `init(forItemsOf:withSearch:)`. Pass `.webMap` as the type and use the search field text to build a search query. Note that webmaps authored prior to July 2nd, 2014, are not supported, so you can also limit the query to only return maps published after that date.
 3. Use `findItems(with:completion:)` to get the first set of matching items.
+
+## Relevant API
+* AGSPortal
```
Comment (philium) [inline arcgis-ios-sdk-samples/Cloud & Portal/Search for webmap by keyword/README.md]:
```suggestion
## Relevant API

* AGSPortal
```

File: arcgis-ios-sdk-samples/Cloud & Portal/Search for webmap by keyword/README.md [hunk 14]
```diff
@@ -1,17 +1,27 @@
...
+* AGSPortalItem
+* AGSPortalQueryParameters
+* AGSPortalQueryResultSet
+
+## Tags
+keyword, query, search, webmap
```
Comment (philium) [inline arcgis-ios-sdk-samples/Cloud & Portal/Search for webmap by keyword/README.md]:
```suggestion
## Tags

keyword, query, search, webmap
```

File: arcgis-ios-sdk-samples/Cloud & Portal/Token authentication/README.md [hunk 15]
```diff
@@ -2,15 +2,15 @@
...
+![Token Authentication sample](token-authentication.png)
 
 ## Use Case
 
 Allows you to access a secure service with the convenience and security of ArcGIS token-based authentication. For example, rather than providing a user name and password every time you want to access a secure service, you only provide those creditials initially to obtain a token which then can be used to access secured resources.
 
 ## How to use the sample
 
-Once you open the sample, you will be challenged for an ArcGIS Online login to view the protected map service. Enter a user name and password for an ArcGIS Online named user account (such as your ArcGIS for Developers account). If you authenticate successfully, the protected map service will display in the map.
+Once you launch the app, you will be challenged for an ArcGIS Online login to view the protected map service. Enter a user name and password for an ArcGIS Online named user account (such as your ArcGIS for Developers account). If you authenticate successfully, the protected map service will display in the map.
```
Comment (philium) [inline arcgis-ios-sdk-samples/Cloud & Portal/Token authentication/README.md]:
```suggestion
Upon opening the sample, you will be challenged for an ArcGIS Online login to view the protected map service. Enter a user name and password for an ArcGIS Online named user account (such as your ArcGIS for Developers account). If you authenticate successfully, the protected map service will display in the map view.
```

File: arcgis-ios-sdk-samples/Content Display Logic/ContentPList.plist [hunk 16]
```diff
@@ -1322,21 +1322,21 @@
 				<key>storyboardName</key>
 				<string>IntegratedWindowsAuthentication</string>
 				<key>descriptionText</key>
-				<string>Uses Windows credentials to access services hosted on a portal secured with Integrated Windows Authentication (IWA).</string>
+				<string>Connect to an IWA secured Portal and search for maps.</string>
```
Comment (philium) [inline arcgis-ios-sdk-samples/Content Display Logic/ContentPList.plist]:
```suggestion
				<string>Connect to an IWA secured portal and search for maps.</string>
```

File: arcgis-ios-sdk-samples/Content Display Logic/ContentPList.plist [hunk 17]
```diff
@@ -1322,21 +1322,21 @@
...
 			</dict>
 			<dict>
 				<key>displayName</key>
 				<string>Authenticate with OAuth</string>
 				<key>descriptionText</key>
-				<string>This sample demonstrates how to authenticate with ArcGIS Online (or your own portal) using OAuth2 to access secured resources (such as private web maps or layers). Accessing secured items requires a login on the portal that hosts them (an ArcGIS Online account, for example).</string>
+				<string>Authenticate with ArcGIS Online (or your own portal) using OAuth2 to access secured resources (such as private web maps or layers).</string>
```
Comment (philium) [inline arcgis-ios-sdk-samples/Content Display Logic/ContentPList.plist]:
```suggestion
				<string>Authenticate with ArcGIS Online (or your own portal) using OAuth2 to access secured resources, such as private web maps or layers.</string>
```

File: arcgis-ios-sdk-samples/Cloud & Portal/Integrated Windows Authentication/README.md (hunk 3 above)
Comment (vquach2404) [reply arcgis-ios-sdk-samples/Cloud & Portal/Integrated Windows Authentication/README.md]: Didn't notice the screenshot! I'll update it.

**Comment by philium** [comment]
@vquach2404 It looks like there are still some headers that are not separated by their content with a blank line. I've suggested the changes, but I'm guessing you may not have seen them because they're hidden:

image

**Comment by vquach2404** [comment]
> @vquach2404 It looks like there are still some headers that are not separated by their content with a blank line. I've suggested the changes, but I'm guessing you may not have seen them because they're hidden:
> 
> image

Oops, thought I got them all. Should be all fixed now!

File: arcgis-ios-sdk-samples/Cloud & Portal/Search for webmap by keyword/README.md [hunk 18]
```diff
@@ -1,17 +1,30 @@
...
+![Search for Webmap by Keyword](search-for-webmap-by-keyword.png)
+
+## Use case
+
+Portals can contain many portal items and, at times, you may wish to query the portal to find what you're looking for. In this example, we search for webmap portal items using a text search.
 
 ## How to use the sample
 
-Input a keyword into the search field. Results load automatically as you type. Tap a result to show the webmap in a map view. 
+Enter search terms into the search bar. Once the search is complete, a list is populated with the resultant webmaps. Tap on a webmap to set it to the map view..
```
Comment (philium) [inline arcgis-ios-sdk-samples/Cloud & Portal/Search for webmap by keyword/README.md]: My bad.

```suggestion
Enter search terms into the search bar. Once the search is complete, a list is populated with the resultant webmaps. Tap on a webmap to set it to the map view.
```

**Pull request merged**